examples/vision/adain.py

```diff
@@ -0,0 +1,675 @@
+            layers.Conv2D(filters=256, **config),
+            layers.UpSampling2D(),
+            layers.Conv2D(filters=128, **config),
+            layers.Conv2D(filters=128, **config),
+            layers.UpSampling2D(),
+            layers.Conv2D(filters=64, **config),
+            layers.Conv2D(
+                filters=3,
+                kernel_size=3,
+                strides=1,
+                padding="same",
+                activation="sigmoid",
+            ),
+        ]
+    )
+    return decoder
+
+
+"""
+## Loss Network
+
+Here we build the loss functions for the neural style transfer model.
+The authors propose to use a pretrained VGG-19 to compute the loss
+function of the network. It is important to keep in mind that this
+will be used for training only the decoder netwrok. The total
+loss (`Lt`) is a weighted combination of content loss (`Lc`) and style
+loss (`Ls`). The `lambda` term is used to vary the amount of style
+transfered.
+
+![The total loss](https://i.imgur.com/Q5y1jUM.png)
+
+### Content Loss
+
+This is the Euclidean distance between the content image features
+and the features of the neural style transferred image.
+
+![The content loss](https://i.imgur.com/dZ0uD0N.png)
+
+Here the authors propose to use the output from the AdaIn layer `t` as
+the content target rather than using features of the original image as
+target. This is done to optimize faster convergence of the loss.
+
+### Style Loss
+
+Rather than using the more commonly used [Gram
+Matrix](https://mathworld.wolfram.com/GramMatrix.html) the authors
+propose to compute the difference between the statistical features
+(mean and variance) which makes it conceptually cleaner. This can
+easily visualized in the following equation
+
+![The style loss](https://i.imgur.com/Ctclhn3.png)
+
+where `theta` denotes the layers in VGG-19 used to compute the loss.
+In this case this corresponds to:
+
+- `block1_conv1`
+- `block1_conv2`
+- `block1_conv3`
+- `block1_conv4`
+
+"""
+
+
+def get_loss_net():
+    vgg19 = keras.applications.VGG19(
+        include_top=False, weights="imagenet", input_shape=(*IMAGE_SIZE, 3)
+    )
+    vgg19.trainable = False
+    layer_names = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1"]
+    outputs = [vgg19.get_layer(name).output for name in layer_names]
+    mini_vgg19 = keras.Model(vgg19.input, outputs)
+
+    inputs = layers.Input([*IMAGE_SIZE, 3])
+    mini_vgg19_out = mini_vgg19(inputs)
+    return keras.Model(inputs, mini_vgg19_out, name="loss_net")
+
+
+"""
+## Neural Style Transfer
+
+This is the trainer module. We wrap the encoder and decoder inside a
+trainer module so that we can harness the beautiful `model.fit` API. 
+"""
+
+
+class NST(tf.keras.Model):
+    def __init__(self, encoder, decoder, ada_in, loss_net, lamb):
+        super().__init__()
+        self.encoder = encoder
+        self.decoder = decoder
+        self.ada_in = ada_in
+        self.loss_net = loss_net
+        self.lamb = lamb
```

Comment:
if this is lambda, let's just call it `lambda`.  The lamb abbreviation can cause confusion.

Reply:
We think ```style_weight``` is more accurate here, so we replaced ```lamb``` with that.